util/db/dbutil.go

```diff
@@ -453,3 +442,49 @@ type idemFn func(ctx context.Context, tx *sql.Tx) error
 
 const infoTxRetries = 5
 const warnTxRetriesInterval = 1
+
+// SynchronousMode is the syncronious modes supported by sqlite database.
+type SynchronousMode int
+
+const (
+	// SynchronousModeOff (0), SQLite continues without syncing as soon as it has handed data off to the operating system. If the application running SQLite crashes,
+	// the data will be safe, but the database might become corrupted if the operating system crashes or the computer loses power before that data has been written to the
+	// disk surface. On the other hand, commits can be orders of magnitude faster with synchronous OFF.
+	SynchronousModeOff SynchronousMode = 0
+	// SynchronousModeNormal (1), the SQLite database engine will still sync at the most critical moments, but less often than in FULL mode. There is a very small
+	// (though non-zero) chance that a power failure at just the wrong time could corrupt the database in journal_mode=DELETE on an older filesystem.
+	// WAL mode is safe from corruption with synchronous=NORMAL, and probably DELETE mode is safe too on modern filesystems. WAL mode is always consistent with synchronous=NORMAL,
+	// but WAL mode does lose durability. A transaction committed in WAL mode with synchronous=NORMAL might roll back following a power loss or system crash.
+	// Transactions are durable across application crashes regardless of the synchronous setting or journal mode.
+	// The synchronous=NORMAL setting is a good choice for most applications running in WAL mode.
+	SynchronousModeNormal SynchronousMode = 1
+	// SynchronousModeFull (2), the SQLite database engine will use the xSync method of the VFS to ensure that all content is safely written to the disk surface prior to continuing.
+	// This ensures that an operating system crash or power failure will not corrupt the database. FULL synchronous is very safe, but it is also slower.
+	// FULL is the most commonly used synchronous setting when not in WAL mode.
+	SynchronousModeFull SynchronousMode = 2
+	// SynchronousModeExtra synchronous is like FULL with the addition that the directory containing a rollback journal is synced after that journal is unlinked to commit a
+	// transaction in DELETE mode. EXTRA provides additional durability if the commit is followed closely by a power loss.
+	SynchronousModeExtra SynchronousMode = 3
+)
+
+// SetSynchronousMode updates the syncronous mode of the connection
+func (db *Accessor) SetSynchronousMode(ctx context.Context, mode SynchronousMode, fullfsync bool) (err error) {
+	if mode < SynchronousModeOff || mode > SynchronousModeExtra {
+		return fmt.Errorf("invalid value(%d) was provided to mode", mode)
+	}
+	_, err = db.Handle.ExecContext(ctx, fmt.Sprintf("PRAGMA synchronous=%d", mode))
+	if err != nil {
+		return err
+	}
+	if fullfsync {
```

Comment:
make this statement just be `if mode >= SynchronousModeFull` ? I think we can drop the argument and simplify the callers.

Reply:
yes, I was thinking about that. I thought that we might want to have the database tier still provide that flexibility in case we'll want it in the future.
( I do realize that currently it's an extra parameter )